app/server/methods/data.js

```diff
@@ -1,10 +1,73 @@
+const async = require("async");
+
+async function _exportDataToConcernedColection(collectionName, data) {
+  if (!Array.isArray(data)) throw "Data not in required format";
+
+  console.log("Calling data for coll: ", collectionName);
+
+  switch (collectionName) {
+    case "activities":
+      return await _insert(Activities.insert, data);
+
+    case "activityTypes":
+      return await _insert(ActivityTypes.insert, data);
+
+    case "feelings":
+      return await _insert(Feelings.insert, data);
+
+    case "groups":
+      return await _insert(Groups.insert, data);
+
+    case "homes":
+      return await _insert(Homes.insert, data);
+
+    case "residents":
+      return await _insert(Residents.insert, data);
+
+    case "roles":
+      return await _insert(Meteor.roles.insert, data);
+
+    case "settings":
+      return await _insert(Settings.insert, data);
+
+    case "users":
+      return await _insert(Meteor.users.insert, data);
+  }
+  function _insert(func, data) {
+    // let insertMethod = Meteor.wrapAsync(func);
+
+    return new Promise((resolve, reject) => {
```

Comment:
We shouldn't need to use promises nor async/await in this solution, as Meteor does that for us.